app/services/internal/remove_stale_contacts_service.rb

```diff
@@ -0,0 +1,43 @@
+class Internal::RemoveStaleContactsService
+  def perform
+    return unless remove_stale_contacts_job_enabled?
+
+    time_period = 30.days.ago
+    contacts_to_delete = stale_contacts(time_period)
+
+    log_stale_contacts_deletion(contacts_to_delete, time_period)
+
+    # Since the number of records to delete is very high,
+    # delete_all would be faster than destroy_all since it operates at database level
+    # and avoid loading all the records in memory
+    # Transaction and batching is used to avoid deadlock and memory issues
+    Contact.transaction do
+      contacts_to_delete
+        .find_in_batches(batch_size: 10_000) do |group|
+          Contact.where(id: group.map(&:id)).delete_all
+        end
+    end
+  end
```

Comment:
Use destroy. limit it to account.contacts. do not use transaction.

Reply:
cool, will make change